File: SqlUtilityLibrary/SqlTypeHelper.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace SqlUtilityLibrary
{
    public static class SqlTypeHelper
    {
        private static readonly Dictionary<Type, SqlDbType> TypeMap;

        // Create and populate the dictionary in the static constructor
        static SqlTypeHelper()
        {
            TypeMap = new Dictionary<Type, SqlDbType>
            {
                [typeof(string)] = SqlDbType.NVarChar,
                [typeof(char[])] = SqlDbType.NVarChar,
                [typeof(byte)] = SqlDbType.TinyInt,
                [typeof(short)] = SqlDbType.SmallInt,
                [typeof(int)] = SqlDbType.Int,
                [typeof(long)] = SqlDbType.BigInt,
                [typeof(byte[])] = SqlDbType.Image,
                [typeof(bool)] = SqlDbType.Bit,
                [typeof(DateTime)] = SqlDbType.DateTime2,
                [typeof(DateTimeOffset)] = SqlDbType.DateTimeOffset,
                [typeof(decimal)] = SqlDbType.Money,
                [typeof(float)] = SqlDbType.Real,
                [typeof(double)] = SqlDbType.Float,
                [typeof(TimeSpan)] = SqlDbType.Time
            };

            /* not in above then added them */
        }

        /// <summary>
        /// Get SqlDbType for givenType
        /// </summary>
        /// <param name="giveType"></param>
        /// <returns><see cref="SqlDbType"/></returns>
        public static SqlDbType GetDbType(Type giveType)
        {
            // Allow nullable types to be handled
            giveType = Nullable.GetUnderlyingType(giveType) ?? giveType;

            if (TypeMap.ContainsKey(giveType))
            {
                return TypeMap[giveType];
            }

            throw new ArgumentException($"{giveType.FullName} is not a supported .NET class");
        }
    }
}

File: QueryBuildLibrary/SqlTypeHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Data

Public Module SqlTypeHelper
    Private ReadOnly TypeMap As Dictionary(Of Type, SqlDbType)

    ' Create and populate the dictionary in the static constructor
    Sub New()
        TypeMap = New Dictionary(Of Type, SqlDbType) From {
            {GetType(String), SqlDbType.NVarChar},
            {GetType(Char()), SqlDbType.NVarChar},
            {GetType(Byte), SqlDbType.TinyInt},
            {GetType(Short), SqlDbType.SmallInt},
            {GetType(Integer), SqlDbType.Int},
            {GetType(Long), SqlDbType.BigInt},
            {GetType(Byte()), SqlDbType.Image},
            {GetType(Boolean), SqlDbType.Bit},
            {GetType(DateTime), SqlDbType.DateTime2},
            {GetType(DateTimeOffset), SqlDbType.DateTimeOffset},
            {GetType(Decimal), SqlDbType.Money},
            {GetType(Single), SqlDbType.Real},
            {GetType(Double), SqlDbType.Float},
            {GetType(TimeSpan), SqlDbType.Time}
            }

        ' not in above then added them 
    End Sub

    ''' <summary>
    ''' Get SqlDbType for givenType
    ''' </summary>
    ''' <param name="giveType"></param>
    ''' <returns><see cref="SqlDbType"/></returns>
    Public Function GetDbType(ByVal giveType As Type) As SqlDbType
        ' Allow nullable types to be handled
        giveType = If(Nullable.GetUnderlyingType(giveType), giveType)

        If TypeMap.ContainsKey(giveType) Then
            Return TypeMap(giveType)
        End If

        Throw New ArgumentException($"{giveType.FullName} is not a supported .NET class")
    End Function
End Module